zcash_proofs/src/circuit/pedersen_hash.rs

```diff
@@ -26,12 +26,12 @@ where
     assert_eq!(personalization.len(), 6);
 
     let mut edwards_result = None;
-    let mut bits = personalization.iter().chain(bits.iter());
+    let mut bits = personalization.iter().chain(bits.iter()).peekable();
     let mut segment_generators = params.pedersen_circuit_generators().iter();
     let boolean_false = Boolean::constant(false);
 
     let mut segment_i = 0;
-    loop {
+    while bits.peek().is_some() {
```

Comment:
This makes the `None` case of `segment_result` unreachable, and I'd like this to be more obvious. We could refactor the inner `while` loop to address this, but it's also probably fine to just call `unreachable!()` instead of `break`.

Reply:
Fixed in commit `Use expect to remove unreachable break.`